src/js/letters/containers/AddressSection.jsx

```diff
@@ -8,26 +8,63 @@ import {
   isDomesticAddress,
   isMilitaryAddress,
   isInternationalAddress,
-  invalidAddressProperty,
   addressUpdateUnavailable
 } from '../utils/helpers.jsx';
-import { updateAddress, saveAddress } from '../actions/letters';
+import { saveAddress } from '../actions/letters';
 import Address from '../components/Address';
+import InvalidAddress from '../components/InvalidAddress';
+import AddressContent from '../components/AddressContent';
 
 export class AddressSection extends React.Component {
   constructor() {
     super();
-    this.state = { isEditingAddress: false };
+    this.state = {
+      hasLoadedAddress: false,
+      isEditingAddress: false,
+      editableAddress: {},
+    };
   }
 
-  handleUpdate = () => {
+  componentWillReceiveProps(nextProps) {
+    if (!this.state.hasLoadedAddress && Object.keys(nextProps.savedAddress).length > 0) {
```

Comment:
We should have put a comment in here when we knew why this was necessary. Can you do that?

If I remember, it was something like...`savedAddress` is coming from an api call that may or may not get returned before this component mounts, so we had to update the editable address when we get it (but only the first time).

Reply:
Yep...not available at time of mounting in most cases, so the user would get a blank form instead of one pre-populated with their existing info. Will add comment

Reply:
Maybe? Depends on how it's done. We'd have to hold off on constructing `<AddressSection/>` until the fetches are finished for it to fix this.

For now, though, I think this is pretty efficient and effective...if also a bit smelly. :man_shrugging: